File: src/main/java/me/spartacus04/jext/command/ParameterLocation.java
package me.spartacus04.jext.command;

import java.util.ArrayList;
import java.util.List;

import org.bukkit.Location;
import org.bukkit.block.Block;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

public class ParameterLocation extends Parameter {

  enum Axis {
    X, Y, Z
  }

  private Axis axis;

  ParameterLocation(boolean required, Axis axis) {
    super(required);
    this.axis = axis;
  }

  @Override
  String getName() {
    return this.axis.toString().toLowerCase();
  }

  @Override
  List<String> onComplete(String parameter, CommandSender sender) {
    final List<String> suggestions = new ArrayList<>();
    suggestions.add("~");
    final Player player = (sender instanceof Player) ? (Player)sender : null;
    if (player == null) return suggestions;
    final Block block = player.getTargetBlockExact(4);
    if (block == null) return suggestions;
    final Location location = block.getLocation();
    switch(axis) {
      case X: {
        Integer a = location.getBlockX();
        suggestions.add(a.toString());
        break;
      }
      case Y: {
        Integer a = location.getBlockY();
        suggestions.add(a.toString());
        break;
      }
      case Z: {
        Integer a = location.getBlockZ();
        suggestions.add(a.toString());
        break;
      }
      default:
        break;
    }
    return suggestions;
  }
  
}

File: src/main/java/me/spartacus04/jext/command/ParameterLocation.kt
package me.spartacus04.jext.command

import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.*

class ParameterLocation internal constructor(required: Boolean, private val axis: Axis) : Parameter(required) {
    internal enum class Axis {
        X, Y, Z
    }

    override val name: String
        get() = axis.toString().lowercase(Locale.getDefault())

    override fun onComplete(parameter: String, sender: CommandSender?): List<String> {
        val suggestions: MutableList<String> = ArrayList()

        suggestions.add("~")

        val player = (if (sender is Player) sender else null) ?: return suggestions.toList()
        val block = player.getTargetBlockExact(4) ?: return suggestions.toList()

        val location = block.location

        when (axis) {
            Axis.X -> {
                val a = location.blockX
                suggestions.add(a.toString())
            }
            Axis.Y -> {
                val a = location.blockY
                suggestions.add(a.toString())
            }
            Axis.Z -> {
                val a = location.blockZ
                suggestions.add(a.toString())
            }
        }
        return suggestions.toList()
    }
}